File: sotraco-backend/app/Models/Ligne.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Ligne extends Model
{
    use HasFactory;

    protected $fillable = ['code', 'nom', 'depart', 'destination', 'couleur', 'description', 'actif'];

    protected $casts = [
        'actif' => 'boolean',
    ];

    public function arrets()
    {
        return $this->belongsToMany(Arret::class, 'ligne_arret')
            ->withPivot('ordre')
            ->orderBy('ligne_arret.ordre');
    }

    public function buses()
    {
        return $this->hasMany(Bus::class);
    }
}

File: sotraco_app/lib/models/ligne.dart
class Ligne {
  final int id;
  final String code;
  final String nom;
  final String? depart;
  final String? destination;
  final String couleur;
  final String? description;
  final int? busesCount;

  Ligne({
    required this.id,
    required this.code,
    required this.nom,
    this.depart,
    this.destination,
    required this.couleur,
    this.description,
    this.busesCount,
  });

  factory Ligne.fromJson(Map<String, dynamic> json) => Ligne(
        id: json['id'],
        code: json['code'],
        nom: json['nom'],
        depart: json['depart'],
        destination: json['destination'],
        couleur: json['couleur'] ?? '#1E824C',
        description: json['description'],
        busesCount: json['buses_count'],
      );
}
